### ISSUE TITLE
test: new end-to-end test suite

### ISSUE THREAD

**Issue by erikgrinaker** [tendermint/tendermint#5291]
We currently have a very limited set of end-to-end tests under `test/p2p`, mainly focused on testing the P2P networking. We've had multiple issues recently that completely broke functionality in obvious ways (e.g. fast sync and state sync) since we don't do whole-system testing. We need more comprehensive end-to-end tests, which should support e.g.:

- Varying configuration and genesis parameters, for single nodes and entire networks
- Different network topologies, including sentry nodes and adding/removing validators
- Simple chaos testing (e.g. disconnecting the network, killing processes)

The suite needs to be able to run permutations of these in parallel, such that we can e.g. have a nightly run with a few hundred networks.

Comprehensive chaos testing (e.g. Jepsen tests) is out of scope for the initial version, and should possibly be implemented separately.

Related to #2805, #3689, #4641, #2795.

____

#### For Admin Use

- [x] Not duplicate issue
- [x] Appropriate labels applied
- [x] Appropriate contributors tagged
- [x] Contributor assigned/self-assigned

**Comment by ebuchman** [comment]
Note we've done some work in this direction with https://github.com/informalsystems/got, https://github.com/informalsystems/testnets, and https://github.com/informalsystems/tm-load-test. We should see if those tools can help address end-to-end automated testing needs here.
